Tools/clinic/clinic.py

```diff
@@ -6174,7 +6176,11 @@ def do_post_block_processing_cleanup(self, lineno: int) -> None:
             return
 
         self.check_remaining_star(lineno)
-        self.function.docstring = self.format_docstring()
+        try:
+            self.function.docstring = self.format_docstring()
+        except ClinicError as exc:
+            exc.lineno = lineno
```

Comment:
Question: why `filename` is not set here?

Reply:
I don't think it is needed; but OTOH, it would not hurt.

Reply:
Added in 3775ab38f90ae447466ee6cdead8777f4c3be29d